src/team_repo.rs

```diff
@@ -0,0 +1,80 @@
+//! The code in this module interacts with the
+//! <https://github.com/rust-lang/team/> repository.
+//!
+//! The [TeamRepo] trait is used to abstract away the HTTP client for testing
+//! purposes. The [TeamRepoImpl] struct is the actual implementation of
+//! the trait.
+
+use crate::certs;
+use async_trait::async_trait;
+use mockall::automock;
+use reqwest::{Certificate, Client};
+
+#[automock]
+#[async_trait]
+pub trait TeamRepo {
+    async fn get_team(&self, name: &str) -> anyhow::Result<Team>;
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct Team {
+    pub name: String,
+    pub kind: String,
+    pub members: Vec<Member>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct Member {
+    pub name: String,
+    pub github: String,
+    pub github_id: i32,
+    pub is_lead: bool,
+}
```

Comment:
there are similar structs in the team and sync-team repos, but I'm not aware of them being published to a public crate. since it's only seven fields across two structs it's probably sufficient for now to have a copy here?

Reply:
it looks like a couple of other projects are depending on https://github.com/rust-lang/team/tree/master/rust_team_data via git, but I'm not particularly keen on adding a git dependency to crates.io

Reply:
That's fine; I'm not keen on that either.

What it _might_ mean is that we should have a common crate that everyone can depend upon (including `rust_team_data`), but I don't think we need to do that today.